internal/controller/resourceset_controller.go

```diff
@@ -296,36 +296,103 @@ func (r *ResourceSetReconciler) checkDependencies(ctx context.Context,
 
 func (r *ResourceSetReconciler) getInputs(ctx context.Context,
```

Comment:
Please add a comment to this function to explain what it does including deduplication